libraries/goclient/goLibrary.go

```diff
@@ -0,0 +1,126 @@
+package goclient
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/intel/afxdp-plugins-for-kubernetes/constants"
+
+	"github.com/intel/afxdp-plugins-for-kubernetes/internal/host"
+	"github.com/intel/afxdp-plugins-for-kubernetes/internal/uds"
+)
+
+var (
+	hWR          uds.Handler
+	hPod         host.Handler
+	busy_timeout int = 20
+	busy_budget  int = 64
+)
+
+const (
+	requestDelay = 100 * time.Millisecond
+)
+
+/*
+Returns the version of our Handshake as a string
+*/
+func GetVersionStr() string {
+	return constants.Uds.Handshake.Version
+}
+
+/*
+Call this function to initialize the library, returns a cleanup function
+*/
+func Init() uds.CleanupFunc {
+	hWR = uds.NewHandler()
+	hPod = host.NewHandler()
+
+	// init uds Handler for reading and writing
+	if err := hWR.Init(constants.Uds.PodPath, constants.Uds.Protocol, constants.Uds.MsgBufSize, constants.Uds.CtlBufSize, 0*time.Second, ""); err != nil {
+		println("Library Error: Error Initialising UDS server: ", err)
+		os.Exit(1)
```

Comment:
remove the os.Exit throughout the code - we don't want to exit the user's app, we want to return an error and let the users' code decide what to do

You can also remove the println and sprintf, etc.

Your println messages could mostly become new errors, e.g.:
`return cleanup, fmt.Errorf("Error initialising UDS server: %v", err)`